### PR TITLE
Improve supervised modeling to handle arbitrary dataset splits

### LINKED ISSUE #80: Allow to specify a sequence of pre-defined train/test splits for supervised models
Can be achieved by expanding constructor signature of LabeledInstanceTrainTestDataset to receive Sequence[LabeledInstanceDataset] for training_set and test_set.

Also need to modify SklearnPredictorOnEmbeddingsScores to handle this and evedesign-server schema specs

### PR DESCRIPTION
**Pull request by thomashopf**

### CODE REVIEW

**Comment by thomashopf** [comment]
@ethane4

File: src/evedesign/models/supervised.py [hunk 1]
```diff
@@ -60,35 +57,26 @@ class SklearnPredictorOnEmbeddingsScores(SupervisedBaseModel, Scorer, MutationSc
...
         use_embeddings: bool = True,
         use_scores: bool = True,
         override_models_for_training: bool = False,
         target_name: str | None = None,
         pooling: Literal["mean", "max"] | None = "mean",
-        cv_folds: int | None = 5,
         batch_size: BatchSize = 128,
-        random_state: int = 42,
-        n_jobs: int = -1,
+        is_classifier: bool = False,
```
Comment (ethane4) [inline src/evedesign/models/supervised.py]: This kind of assumes that the model outputs themselves are binarized, right?

ex. a classifier trained on binary labels that outputs continuous probabilities will fail to calculate MCC if the predictions are not converted to binary labels

I didn't catch any binarization cutoff param anywhere in here, could be missing it though.

File: src/evedesign/dataset.py [hunk 2]
```diff
@@ -9,13 +11,15 @@ class LabeledInstanceDataset:
 
     Missing labels must be encoded with None (do not use NaN as this will break JSON serialization)
 
-    # TODO: add utility method to create from dataframe
-    # TODO: add utility method for creating train/test split
+    TODO: add utility method to create from dataframe/pgdata
```
Comment (ethane4) [inline src/evedesign/dataset.py]: guessing this would just call to that other fxn in proteingym.py?

File: src/evedesign/proteingym.py [hunk 3]
```diff
@@ -193,7 +193,7 @@ def dataset_to_evedesign(
     split: str | None,
     target: str,
     test_fold: int | None,
-) -> tuple[System, LabeledInstanceTrainTestDataset]:
+) -> tuple[System, LabeledInstanceDataset | None, LabeledInstanceDataset | None]:
```
Comment (ethane4) [inline src/evedesign/proteingym.py:196]: grouping all folds != test fold into train should be fine for the current pg-benchmark (I believe CV runs will spin up n jobs for n folds)

Still means we will need to figure something out for models that take validation down the line... May be good to output a four-element tuple w/ system, train instance dataset, test instance dataset, val instance dataset?

Comment (ethane4) [reply src/evedesign/proteingym.py:196]: And val can just be None most of the time

File: src/evedesign/models/supervised.py (hunk 1 above)
Comment (thomashopf) [reply src/evedesign/models/supervised.py]: oh really good catch... the assumption for classifiers is to output binary classes, which the current scikit-learn wrapper will do, so all good for MCC, but it will be an issue for AUROC

also might be nice to actually be able to return the probabilities instead of the classes

File: src/evedesign/dataset.py (hunk 2 above)
Comment (thomashopf) [reply src/evedesign/dataset.py]: was thinking about being able to use the dataset outside a dvc pipeline, so with full dataset loaded and splits attached to the dataset, but I suppose we are good with the existing function in proteingym.py and don't need to create extra coupling here

File: src/evedesign/proteingym.py (hunk 3 above)
Comment (thomashopf) [reply src/evedesign/proteingym.py:196]: Yep that functionality is currently still missing from PR... my working model is to attach the train/val split to the training set to it can be passed to build() as data argument, so won't need to update anything here

File: src/evedesign/models/supervised.py (hunk 1 above)
Comment (thomashopf) [reply src/evedesign/models/supervised.py]: returning log probabilities from classifiers is a much better solution actually that very cleanly integrates overall - goodbye MCC ;)
